release_dashboard/lib/widgets/common/dialog_prompt.dart

```diff
@@ -4,37 +4,115 @@
 
 import 'package:flutter/material.dart';
 
-/// Widget that prompts an alert dialogue to the current window and forces the user to choose between two options.
+typedef DynamicFuture = Future<dynamic> Function();
+
+/// Function that prompts an alert to the current window and forces the user to choose between two options.
 Future<String?> dialogPrompt({
   required BuildContext context,
-  required String title,
-  required String content,
-  required String leftOptionTitle,
-  required String rightOptionTitle,
-  VoidCallback? leftOptionCallback,
-  VoidCallback? rightOptionCallback,
+  required Widget title,
+  required Widget? content,
+  required String leftButtonTitle,
+  required String rightButtonTitle,
+  DynamicFuture? leftButtonCallback,
+  DynamicFuture? rightButtonCallback,
 }) {
   return showDialog<String>(
     context: context,
     builder: (BuildContext context) => AlertDialog(
-      title: Text(title),
-      content: Text(content),
+      title: title,
+      content: content,
       actions: <Widget>[
         TextButton(
-          onPressed: () {
-            if (leftOptionCallback != null) leftOptionCallback();
-            Navigator.pop(context, leftOptionTitle);
+          onPressed: () async {
+            if (leftButtonCallback != null) await leftButtonCallback();
+            Navigator.pop(context, leftButtonTitle);
           },
-          child: Text(leftOptionTitle),
+          child: Text(leftButtonTitle),
         ),
         TextButton(
-          onPressed: () {
-            if (rightOptionCallback != null) rightOptionCallback();
-            Navigator.pop(context, rightOptionTitle);
+          onPressed: () async {
+            if (rightButtonCallback != null) await rightButtonCallback();
```

Comment:
Await'ing here will hold the rendering pipeline.

Using something like https://pub.dev/packages/progress_state_button will display some progress until the future is completed.

Reply:
I can add it when tackling this issue: https://github.com/flutter/flutter/issues/93750

Although, the `conductor clean` command is very fast. The progress button will likely not be visible at all.

Reply:
It's a best practice. There'll be more longer async processes like `startContext` and `publishRelease` where we'll want this, but sgtm to defer.

Let's add a tracking TODO here so it's easy to add in the future.

Reply:
Done